groups/views.py

```diff
@@ -34,6 +39,39 @@ def dispatch(self, request, *args, **kwargs):
         return super(GroupDetail, self).dispatch(request, *args, **kwargs)
 
 
+class SubscribeBase(UpdateView):
+    def get_form_kwargs(self):
+        """Pass the user to the form to check subscription state."""
+        kwargs = super(SubscribeBase, self).get_form_kwargs()
+        kwargs.update({'user': self.request.user})
+        return kwargs
+
+    def form_valid(self, form):
+        """Subscribe or unsubscribe the request user."""
+        user = self.request.user
+
+        if form.initial['subscribe']:
+            self.add(user)
+        else:
+            self.remove(user)
+
+        return super(SubscribeBase, self).form_valid(form)
```

Comment:
I don't think that this needs to be an `UpdateView` -- that's not a big deal, I guess.

I do however seriously think that calling `super` (and thus `self.object.save()` here is Wrong™. That can have side effects that we do not intend.